스토어 상세 UI [#16]

juyeonnnn opened the pull request:
# ✨ Feature - 스토어 상세 페이지 UI 구현

<img width="402" height="856" alt="Image" src="https://github.com/user-attachments/assets/9f2565e2-5dff-438a-a604-8a4f2c909a0c" />

### 📝 Description  
-  스토어 상세 페이지 UI 구현

<img width="380" height="830" alt="Screenshot_1753266762" src="https://github.com/user-attachments/assets/eef3315a-7cab-46e9-863e-8393362a761c" />

---

### ✅ To Do List  
- [x] 가게 정보 
- [x] 가게 리뷰
- [x] 리뷰 슬라이드 구현 
- [x] 리뷰 카드 그림자 추가
 
---

### 📚 Etc  
없음 


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **신규 기능**
  * 매장 상세 정보를 보여주는 화면이 추가되었습니다.
  * 리뷰 카드 컴포넌트가 도입되어 매장 리뷰를 시각적으로 확인할 수 있습니다.
  * 매장 카드에서 해당 매장 상세 페이지로 이동할 수 있는 기능이 추가되었습니다.
  * 매장 검색 기능이 새로운 경로(`/store/search`)로 이동하도록 업데이트되었습니다.

* **버그 수정**
  * 검색 바를 눌렀을 때, 새로운 경로로 이동하도록 동작이 변경되었습니다.

* **기타**
  * 새로운 스타일이 여러 컴포넌트에 적용되었습니다.
  * 신규 패키지(`expo-linear-gradient`)가 추가되었습니다.
  * 불필요한 파일이 삭제되고, 일부 경로가 변경되었습니다.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

juyeonnnn committed:
feat: 스토어 목록 바텀시트 UI 및 검색바, 카테고리 탭 구현 (#3)

juyeonnnn committed:
style: 카테고리 탭 및 카드 뱃지 svg 수정 (#3)

juyeonnnn committed:
feat: 검색 페이지 이동 및 카테고리 탭 필터링 기능 구현

- 검색바 클릭 시 SearchScreen으로 이동
- 좌측 버튼 3개 추가
- 카테고리 탭 선택에 따라 목록 필터링 적용 (#3)

juyeonnnn committed:
fix: 카테고리 이름 '운동과건강' → '운동/건강' 으로 통일 (#3)

juyeonnnn committed:
fix: kkLoginTextbox width는 하드코딩 제거 및 serchbar->small 변경  (#3)

juyeonnnn committed:
fix: layout 괄호 에러 수정 (#3)

juyeonnnn committed:
Merge branch 'develop' into feat/store

juyeonnnn committed:
feat: 스토어 상세 페이지 초기 세팅 (#16)

juyeonnnn committed:
fix:충돌해결

juyeonnnn committed:
feat: 스토어 상세 페이지 UI 완성 (#16)

juyeonnnn committed:
fix: images/store 제거 (#16)

onebone commented on src/assets/images/back-arrow.svg:
이거 #19 에서도 추가된 아이콘이라서 나중에 합쳐야 할 것 같네요!

juyeonnnn replied:
이름 수정했습니다~

onebone commented:
@coderabbit full review

chae1125 commented on src/screens/Store/StoreDetailScreen/StoreDetailScreen.tsx:
```diff
@@ -0,0 +1,130 @@
+
+          <Text style={styles.reviewCount}>(리뷰 {store.reviewCount}개)</Text>
+        </View>
+        <View style={styles.detailSection}>
```
이 디테일 Text 부분에 Pretendard 폰트가 적용이 안 되어 있는 거 같습니다!

juyeonnnn replied:
폰트는 나중에 한다구 들어서 일단 폰트 없이 작업하고 있었는데 추가할까요?

juyeonnnn replied:
수정했습니다!

onebone commented on src/screens/Store/StoreDetailScreen/StoreDetailScreen.style.ts:
```diff
@@ -0,0 +1,98 @@
+
+  backButton: {
+    position: 'absolute',
+    top: 50,
```
50은 피그마에서 상태바 크기를 포함한 크기일까요? 기기마다 상태바 크기가 다를 수 있어서 safe-area의 inset 관련된 정보를 얻어와서 top을 줘야 할 것 같네요!

juyeonnnn replied:
넹 수정했습니다!

juyeonnnn commented on src/screens/Store/StoreDetailScreen/StoreDetailScreen.tsx:
```diff
@@ -0,0 +1,130 @@
+  const router = useRouter();
+  const { id } = useLocalSearchParams();
+
+  const store = mockStores.find((s) => s.id === id);
+  // 가게 정보 없을때 임시 화면
+  if (!store) {
+    return <Text>가게 정보를 불러올 수 없습니다.</Text>;
+  }
+
+  const [isLiked, setIsLiked] = useState(store.isLiked);
+
```
수정 완료~

juyeonnnn committed:
feat: UI 및 KkTextbox 변경사항 적용
- backButton top 위치 safe-area 대응
- 폰트 스타일 추가
- KK 버튼 변경사항 적용
- React Hooks 규칙 위반 수정 useState 최상위 호출  (#21)

juyeonnnn committed:
style: reveiw card 그림자 추가 (#21)

juyeonnnn commented on src/screens/Store/ReviewCard/ReviewCard.style.ts:
```diff
@@ -0,0 +1,52 @@
+    height: 57,
+    width: '100%',
+  },
+  textContainer: {
+    position: 'absolute',
+    top: 120,
+    left: 11,
+    right: 11,
+  },
```
피그마에서 top 120으로 설정해놓으셔서 이렇게 해야할 것 같아

juyeonnnn commented on src/screens/Store/ReviewCard/ReviewCard.style.ts:
```diff
@@ -0,0 +1,52 @@
+import colors from '../../../design/colors';
+
+export default StyleSheet.create({
+  shadowWrapper: {
+    marginBottom: 10,
+    borderRadius: 20,
+    shadowColor: 'rgb(108, 49, 49)',
+    shadowOpacity: 0.15,
+    shadowOffset: { width: 1.02, height: 3.07 },
+    shadowRadius: 3,
+    elevation: 4,
+    backgroundColor: colors.light.white, // 그림자 표시용 배경색
+  },
```
오 고마웡

juyeonnnn committed:
style: KkTextbox width 수정 및 디자인 시스템 그림자 색상 정의 추가 (#21)
